airwallex/src/main/java/com/airwallex/android/view/PaymentMethodsViewModel.kt

```diff
@@ -71,22 +71,19 @@ internal class PaymentMethodsViewModel(
                 )
             }
             is AirwallexRecurringSession -> {
+                @Suppress("TooGenericExceptionCaught")
                 try {
-                    ClientSecretRepository.getInstance().retrieveClientSecret(
-                        requireNotNull(session.customerId),
-                        object : ClientSecretRepository.ClientSecretRetrieveListener {
-                            override fun onClientSecretRetrieve(clientSecret: ClientSecret) {
-                                retrieveAvailablePaymentMethods(
-                                    resultData = resultData,
-                                    clientSecret = clientSecret.value
-                                )
-                            }
-
-                            override fun onClientSecretError(errorMessage: String) {
-                                resultData.value =
-                                    Result.failure(AirwallexCheckoutException(message = errorMessage))
-                            }
-                        }
+                    val clientSecret = try {
+                        ClientSecretRepository.getInstance()
+                            .retrieveClientSecret(requireNotNull(session.customerId))
+                    } catch (e: Exception) {
```

Comment:
You might need to use Throwable unless you can be sure all exceptions are subclasses of Exception.

Reply:
Actually, can we merge this try catch with the outer try catch?

Reply:
Good point, I've made the upstream `AirwallexCheckoutException` too so that these two catching here can be merged.